lib/date_extractors/heic_video_extractor.dart

```diff
@@ -0,0 +1,44 @@
+import 'dart:io';
+import 'package:gpth/date_extractors/json_extractor.dart';
+import "package:path/path.dart" as p;
+
+/// Finds corresponding json file with info and gets 'photoTakenTime' from it
+Future<DateTime?> heicVideoExtractor(File file, {bool tryhard = false}) async {
+  final fileNameWithoutExtension = p.basenameWithoutExtension(file.path);
+
+  if (fileNameWithoutExtension.startsWith("IMG") &&
+      p.extension(file.path).toLowerCase() == ".mp4") {
+    final String parentPath = file.parent.path;
+    // Is a valid iOS Live Photo video
+    return tryGetDateTimeFromPossibleFileExtensions(
+        parentPath, fileNameWithoutExtension);
+  }
+
+  return null;
+}
+
+Future<DateTime?> tryGetDateTimeFromPossibleFileExtensions(
+    String parentPath, String fileNameWithoutExtension,
+    {bool tryhard = false}) async {
+  final possibleFileExtensions = <String>[
+    "HEIC",
+    "heic",
+    "JPG",
+    "jpg",
+    "JPEG",
+    "jpeg"
+  ];
```

Comment:
I have come across a few situations when the original image in the live photo pair is saved as a `.jpg` or `.jpeg`, they seem to use both upper and lower case. I suspect this is either because the HEIC format is disabled or the photo with the live photo feature was sent over iMessage which converted the image into a `.jpeg` format.

Sample directory of the files:
- `IMG_3984.jpeg`
- `IMG_3984.MP4`
- `IMG_3984.jpeg.json`

and...
- `IMG_7334.JPG`
- `IMG_7334.MP4`
- `IMG_7334.JPG.json`